mobile/android/android-emulator-services.ts

```diff
@@ -33,17 +39,34 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 	}
 
 	public checkDependencies(): IFuture<void> {
-		var future = new Future<void>();
-		var proc = this.$childProcess.spawn('emulator', ['-help']);
-		proc.on("error", (args:any) => {
-			this.$errors.fail("The Android SDK is not configured properly. " +
-				"Verify that you have installed the Android SDK and that you have added its `platform-tools` and `tools` directories to your PATH environment variable.");
-		});
-		proc.on("exit", (code:number) => {
-			future.return();
-		});
+		return (() => {
+			this.checkAndroidSDKConfiguration().wait();
+			if(options.geny) {
+				this.checkGenymotionConfiguration().wait();
+			}
+		}).future<void>()();
+	}
 
-		return future;
+	private checkAndroidSDKConfiguration(): IFuture<void> {
+		return (() => {
+			var proc = this.$childProcess.spawnFromEvent('emulator', ['-help'], "exit", undefined, { throwError: false }).wait();
+
+			if(proc.stderr) {
+				this.$errors.fail("The Android SDK is not configured properly. " +
+					"Verify that you have installed the Android SDK and that you have added its `platform-tools` and `tools` directories to your PATH environment variable.");
+			}
+		}).future<void>()();
+	}
+
+	private checkGenymotionConfiguration(): IFuture<void> {
+		return (() => {
+			var proc = this.$childProcess.spawnFromEvent("genyshell", ["-h"], "exit", undefined, { throwError: false }).wait();
+
+			if(proc.stderr) {
+				this.$errors.fail("Genymotion is not configured properly. " +
+					"Verify that you have installed Genymotion and that you have added its installation directory to your PATH environment variable.");
```

Comment:
Do we need the help of the command printed here? It looks like the command line is OK but the dependency is broken.